aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -208,6 +209,14 @@
    */
   public static final String CLIENT_FACTORY = "client.factory";
 
+  /**
+   * The implementation class of {@link software.amazon.awssdk.http.SdkHttpClient} used by {@link AwsClientFactory}
+   * If set, all AWS clients will use this specified HTTP client.
+   * If not set, CLIENT_HTTP_TYPE_DEFAULT will be used.
+   */
+  public static final String CLIENT_HTTP_TYPE = "client.http.type";
+  public static final String CLIENT_HTTP_TYPE_DEFAULT = UrlConnectionHttpClient.class.getName();
```

Comment:
we need to use string here, otherwise it will force loading the UrlConnectionHttpClient class.

Reply:
also because we are talking about "type" but not "impl" here, prefer to use shorter strings instead of full class name. In this case, 2 types could be "url" and "apache"

Reply:
fixed. thanks!